docs/examples/vectorfields.py

```diff
@@ -0,0 +1,493 @@
+import torch
+
+torch.manual_seed(123)
+import jax.random as jr
+import jax.numpy as jnp
+from jaxtyping import install_import_hook
+import matplotlib.pyplot as plt
+from matplotlib import rcParams
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+import numpy as onp
+import optax as ox
+import tensorflow_probability as tfp
+from jax.config import config
+from dataclasses import dataclass
+from jaxtyping import (
+    Array,
+    Float,
+    install_import_hook,
+)
+from jax import hessian, jit
+
+with install_import_hook("gpjax", "beartype.beartype"):
+    import gpjax as gpx
+
+# Enable Float64 for more stable matrix inversions.
+key = jr.PRNGKey(123)
+plt.style.use(
+    "https://raw.githubusercontent.com/JaxGaussianProcesses/GPJax/main/docs/examples/gpjax.mplstyle"
+)
+cols = rcParams["axes.prop_cycle"].by_key()["color"]
+
+# jnp.set_printoptions(4, edgeitems=30, linewidth=100000,
+#     formatter=dict(float=lambda x: "%.3g" % x))
+
+
+# %% [markdown]
+# ## Data Loading
+# Data and original code are publicly available at https://github.com/renatoberlinghieri/Helmholtz-GP
+#
+# The data are 297 (artificial) measurements of ocean velocity, given by the vector field
+#
+# $$
+# \mathbf{F}(\mathbf{x}) = -x^{(1)}\hat{\imath} + x^{(0)}\hat{\jmath},
+# $$
+#
+#
+#
+# where $\mathbf{x} = (x^{(0)}$,$x^{(1)})^\text{T}$, $\hat{\imath}$ and $\hat{\jmath}$ are unit vectors in the standard Cartesian directions (dimensions will be indicated by superscripts).
+#
+# The measurements are contained in a dataset $D_0=\left\{ \left(\mathbf{x}_{0,i} , \mathbf{y}_{0,i} \right)\right\}_{i=1}^N$, where $\mathbf{y}_i$ is a 2 dimensional velocity vector and $\mathbf{x}_i$ is a 2 dimensional position vector on an $N=17\times17$ grid, equally spaced over the interval $[-1,1] \times[-1,1]$. 8 measurements of $\mathbf{F}(\mathbf{x})$ at new locations are contained in a dataset $D_T=\left\{\left(\mathbf{x}_{T,i}, \mathbf{y}_{T,i} \right)\right\}_{i=1}^8$, which are allocated to training data.
+#
+#
+
+# %%
+# loading in data
+try:
+    Pos_Train = onp.genfromtxt("data/XY_train_vortex.csv", delimiter=",").T
+    Pos_Test = onp.genfromtxt("data/XY_test_vortex.csv", delimiter=",").T
+    Vel_Train = onp.genfromtxt("data/UV_train_vortex.csv", delimiter=",").T
+    Vel_Test = onp.genfromtxt("data/UV_test_vortex.csv", delimiter=",").T
+except FileNotFoundError:
+    Pos_Train = onp.genfromtxt(
+        "docs/examples/data/XY_train_vortex.csv", delimiter=","
+    ).T
+    Pos_Test = onp.genfromtxt("docs/examples/data/XY_test_vortex.csv", delimiter=",").T
+    Vel_Train = onp.genfromtxt(
+        "docs/examples/data/UV_train_vortex.csv", delimiter=","
+    ).T
+    Vel_Test = onp.genfromtxt("docs/examples/data/UV_test_vortex.csv", delimiter=",").T
+
+
+fig, ax = plt.subplots(1, 1)
+ax.quiver(Pos_Test[0], Pos_Test[1], Vel_Test[0], Vel_Test[1], label="$D_0$")
+ax.quiver(
+    Pos_Train[0],
+    Pos_Train[1],
+    Vel_Train[0],
+    Vel_Train[1],
+    color="red",
```

Comment:
When specifying colours, please index `cols`. For `"red"`, this will be `cols[1]`. This ensures consistent style across all notebooks.